osp_scraper/spiders/uga.py

```
@@ -0,0 +1,88 @@
+            anchor = dept.css("::text").extract_first()
+            yield scrapy.FormRequest.from_response(
+                response,
+                method="POST",
+                formdata={
+                    'RadioButtonList1': "DEP1",
+                    'ddlDepartment': value
+                },
+                meta={
+                    'depth': 2,
+                    'hops_from_seed': 2,
+                    'source_url': response.url,
+                    'source_anchor': anchor
+                },
+                dont_click=True,
+                callback=self.parse_for_courses
+            )
+
+    def parse_for_courses(self, response):
+        # Grab the search result page.
+        for item in self.parse_for_files(response):
+            yield item
+
+        # Syllabi are stored at the same URL behind javascript links that feed
+        # into ASP.NET POST requests.  So, rather than implement extract_links,
+        # we yield the response of each POST request into parse_for_files.
+        rows = response.css("#gridFileList tr")[1:]
+        for row in rows:
+            link = row.css("td:nth-child(6) a")
+            if link:
+                # Links look like this:
+                # javascript:__doPostBack('<eventtarget>','<eventargument>')
+                eventtarget, eventargument = link.css("::attr(href)")\
+                                                 .re(r"'(.*?)'")
+                anchor = response.meta['source_anchor']
+                anchor += " ".join(row.css("td::text").extract()[:3])
+                anchor += " " + link.css("::text").extract_first()
+                yield scrapy.FormRequest.from_response(
+                    response,
+                    method="POST",
+                    formdata={
+                        '__EVENTTARGET': eventtarget,
+                        '__EVENTARGUMENT': eventargument,
+                        'RadioButtonList1': "DEP1",
+                        'ddlDepartment': "AFS"
```

Oops, that's really bad.  A hardcoded value like that would tend to cause errors.

What's weird is that the errors only come up following some requests, sometimes.  There doesn't seem to be too much consistency.